src/pathDetails/PathDetails.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useEffect, useRef, useState } from 'react'
+import styles from '@/pathDetails/PathDetails.module.css'
+import { HeightGraph } from 'leaflet.heightgraph/src/heightgraph'
+import 'leaflet.heightgraph/src/heightgraph.css'
+import { Path } from '@/api/graphhopper'
+import Dispatcher from '@/stores/Dispatcher'
+import { PathDetailsElevationSelected, PathDetailsHover, PathDetailsRangeSelected } from '@/actions/Actions'
+import { Coordinate } from '@/stores/QueryStore'
+
+interface PathDetailsProps {
+    selectedPath: Path
+}
+
+export default function ({ selectedPath }: PathDetailsProps) {
+    const containerRef: React.RefObject<HTMLDivElement> = useRef(null)
+    const [graph, setGraph] = useState<any | null>(null)
+    useEffect(() => {
+        const options = {
+            width: containerRef.current!.clientWidth,
+            height: containerRef.current!.clientHeight,
+            // todo: since we do not use this maybe we can/should remove the svg asset rules again because this we added
+            //       them just for this...
+            expandControls: false,
+        }
+        const callbacks = {
+            pointSelectedCallback: onPathDetailHover,
+            areaSelectedCallback: onRangeSelected,
+            routeSegmentsSelectedCallback: onElevationSelected,
+        }
+        setGraph(new HeightGraph(containerRef.current, options, callbacks))
+    }, [containerRef])
```

Comment:
Is the old heightgraph properly removed from the dom? I.e. when the large and small layouts are exhchanged this component will be unmounted. I don't know from just looking at this. 

When the dependeny array is removed there are a lot of svg added to the dom which maybe indicate a missing cancel function as return value of this effect?

Reply:
The height graph is just a collection of <svg> elements attached to some parent <div>, I think. So when we remove the div the heightgraph is also gone?! But no idea tbh. I saw we can use some clean up function as return value of `useEffect`, but I am not sure what to remove?

Reply:
It should be gone and I just noticed it when removing the dependencies as long as there is the containerRef dependency, I don't see more svg elements than  expected.

Reply:
You mean there is nothing we need to do? :)

Reply:
Yes, sorry about that comment. I typed it while thinking, not considering whether it would be helpful or not.